src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/EETypeOptionalFieldsNode.cs

```diff
@@ -19,7 +19,7 @@ public override ObjectNodeSection GetSection(NodeFactory factory)
             if (factory.Target.IsWindows)
```

Comment:
Should we extract this piece of logic into a helper method on `ObjectNodeSection` or `NodeFactory` so that we do not need to fix up a bunch of places everytime a new right way to do foldable readonly data sections shows up?